components/dashboards-web-component/src/designer/components/WidgetOptionsConfiguration.jsx

```diff
@@ -17,28 +17,143 @@
  */
 
 import React, { Component } from 'react';
+
+import { FormattedMessage } from 'react-intl';
+
 import PropTypes from 'prop-types';
 import _ from 'lodash';
+import { Card, CardHeader, CardText, Checkbox, MenuItem, SelectField, TextField } from 'material-ui';
+
+const labelStyle = {
+    fontSize: 12,
+};
 
 export default class WidgetOptionsConfiguration extends Component {
     constructor(props) {
         super(props);
-
         this.doesSelectedWidgetHasOptions = this.doesSelectedWidgetHasOptions.bind(this);
+        this.getCurrentWidgetOptionsInputs = this.getCurrentWidgetOptionsInputs.bind(this);
+        this.handleWidgetOutput = this.handleWidgetOutput.bind(this);
+        this.state = {
+            options: props.selectedWidget.props.configs.options,
+        };
+    }
+
+    getCurrentWidgetOptionsInputs() {
+        const CurrentWidgetOptionsInputs = [];
+        const options = this.props.selectedWidgetConfiguration.configs.options;
+        if (options) {
+            for (let i = 0; i < options.length; i++) {
+                const value = this.state.options[options[i].id];
+                switch (options[i].type.name.toUpperCase()) {
```

Comment:
Lets cache `options[i]` so that it is evaluated only once. `const currentOption = options[i]`